src/axom/quest/MarchingCubes.cpp

```diff
@@ -59,184 +97,181 @@ void MarchingCubes::setFunctionField(const std::string& fcnField)
+                                             facetNodeCoordsView,
+                                             facetParentIdsView,
+                                             m_facetIndexOffsets[d]);
+  }
+
+  for(axom::IndexType d = 0; d < m_singles.size(); ++d)
+  {
+    m_singles[d]->computeFacets();
+  }
+
+  for(axom::IndexType d = 0; d < m_singles.size(); ++d)
   {
-    contourCellCount += m_singles[dId]->getContourCellCount();
+    const auto domainId = m_singles[d]->getDomainId(d);
+    const auto domainFacetCount =
+      (d < m_singles.size() - 1 ? m_facetIndexOffsets[d + 1] : m_facetCount) -
+      m_facetIndexOffsets[d];
+    m_facetDomainIds.fill(domainId, domainFacetCount, m_facetIndexOffsets[d]);
   }
-  return contourCellCount;
 }
 
 axom::IndexType MarchingCubes::getContourNodeCount() const
 {
-  axom::IndexType contourNodeCount = 0;
-  for(int dId = 0; dId < m_singles.size(); ++dId)
+  axom::IndexType contourNodeCount =
+    m_singles.empty() ? 0 : m_facetCount * m_singles[0]->spatialDimension();
+  return contourNodeCount;
+}
+
+void MarchingCubes::clearMesh()
+{
+  for(int d = 0; d < m_singles.size(); ++d)
   {
-    contourNodeCount += m_singles[dId]->getContourNodeCount();
+    m_singles[d]->getImpl().clearDomain();
   }
-  return contourNodeCount;
+  m_domainCount = 0;
+  m_facetNodeIds.clear();
+  m_facetNodeCoords.clear();
+  m_facetParentIds.clear();
+}
+
+void MarchingCubes::clearOutput()
+{
+  m_facetCount = 0;
+  m_facetNodeIds.clear();
+  m_facetNodeCoords.clear();
+  m_facetParentIds.clear();
+  m_facetDomainIds.clear();
 }
 
 void MarchingCubes::populateContourMesh(
   axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE>& mesh,
   const std::string& cellIdField,
-  const std::string& domainIdField)
+  const std::string& domainIdField) const
 {
+  AXOM_PERF_MARK_FUNCTION("MarchingCubes::populateContourMesh");
   if(!cellIdField.empty() &&
      !mesh.hasField(cellIdField, axom::mint::CELL_CENTERED))
   {
-    mesh.createField<axom::IndexType>(cellIdField,
-                                      axom::mint::CELL_CENTERED,
-                                      mesh.getDimension());
+    // Create cellId field, currently the multidimensional index of the parent cell.
+    mesh.createField<axom::IndexType>(cellIdField, axom::mint::CELL_CENTERED);
   }
 
   if(!domainIdField.empty() &&
      !mesh.hasField(domainIdField, axom::mint::CELL_CENTERED))
   {
-    mesh.createField<axom::IndexType>(domainIdField, axom::mint::CELL_CENTERED);
+    mesh.createField<DomainIdType>(domainIdField, axom::mint::CELL_CENTERED);
   }
 
   // Reserve space once for all local domains.
   const axom::IndexType contourCellCount = getContourCellCount();
   const axom::IndexType contourNodeCount = getContourNodeCount();
+  // Temporarily disable reservation due to unknown bug.
+  // See https://github.com/LLNL/axom/pull/1271
   mesh.reserveCells(contourCellCount);
   mesh.reserveNodes(contourNodeCount);
```

Comment:
Is this comment still correct?  I see you're calling reserve methods.

Reply:
Removing the comment.  I was able to significantly narrow down the conditions leading to the bug.  Allocating memory fails only in the docker environment configured with umpire and the mesh uses sidre.